administrator/components/com_joomgallery/views/images/view.html.php

```diff
@@ -89,11 +89,10 @@ protected function addToolbar()
       JToolbarHelper::divider();
     }
 
-    //if($canDo->get('core.delete'))
-    //{
+    if($canDo->get('core.delete') || $canDo->get('joom.delete.own'))
```

Comment:
I can imagine why the "if query" was commented out. 
It might be possible that the permission to delete or delete own images is not allowed on component level but e.g. allowed for categories. That would mean, when checking for "core.delete" and "core.delete.own", that the "Delete" button would not be shown although there are images that are allowed to be deleted. What do you think?

By the way, in my opinion the same applies to the queries above (core.edit, core.edit.own, core.edit.state) !? What do you think?

Reply:
>I can imagine why the "if query" was commented out...

You are right, if the authorization is not allowed on a global level, the buttons will not be displayed, even if authorization has been granted for the individual image.
I have commented out the "if query" again now. 

>By the way, in my opinion the same applies to the queries above (core.edit, core.edit.own, core.edit.state) !? What do you think?

I think nobody noticed or bothered that the buttons are missing here, because editing is still possible by clicking on the title.
Would you also comment out the "if query" here?

Reply:
> You are right, if the authorization is not allowed on a global level, the buttons will not be displayed, even if authorization has been granted for the individual image.
I have commented out the "if query" again now.

I would code the following:
```
    if($this->pagination->total)
    {
      JToolbarHelper::deleteList('', 'remove');
    }
```
Than only the button will be displayed if there is at least one image in the list.

>I think nobody noticed or bothered that the buttons are missing here, because editing is still possible by clicking on the title.
Would you also comment out the "if query" here?

The queries for "core.edit / core.edit.own" and "core.edit.state" are not correct because they do not query the authorization of the image but that of the component.

Actually, for each image in the list it should be checked whether the respective authorization is available. Then one can make a correct decision as to whether the respective button has to be displayed or not. Maybe a separate issue? I leave that up to you ;-).

Reply:
>Maybe a separate issue?

Yes, I think the display of the buttons is a separate issue.
Although I do not necessarily consider it necessary to check the authorization for each image in the backend. Probably this would negatively affect the performance.
If the action is carried out, it will be checked anyway whether the necessary permission exists. 

Now I have implemented your suggestion for the delete button, as well as a small change to show the new button.

Thanks again.

Reply:
>Although I do not necessarily consider it necessary to check the authorization for each image in the backend. Probably this would negatively affect the performance.
If the action is carried out, it will be checked anyway whether the necessary permission exists.

I agree.

>Now I have implemented your suggestion for the delete button, as well as a small change to show the new button.

👍 👍